apps/predictions/lib/stream_worker.ex

```diff
@@ -18,44 +17,38 @@ defmodule Predictions.StreamSupervisor.Worker do
   end
 
   @impl Supervisor
-  def init(key) do
-    sses_stream_name = sses_stream_name(key)
-    api_stream_name = :"predictions_api_stream_#{key}"
-    prediction_stream_name = :"predictions_data_stream_#{key}"
+  @doc "The `filters` argument is a string representing the filter parameters of
+  the resultant API call e.g. 'filter[route]=CR-Foxboro&filter[direction_id]=0'"
+  def init(filters) do
+    sses_stream_name = sses_stream_name(filters)
+    api_stream_name = :"predictions_api_stream_#{filters}"
+    prediction_stream_name = :"predictions_data_stream_#{filters}"
 
     Supervisor.init(
       [
-        {ServerSentEventStage, sses_opts(key)},
+        {ServerSentEventStage, sses_opts(filters)},
         {V3Api.Stream, name: api_stream_name, subscribe_to: sses_stream_name},
         {Predictions.Stream, name: prediction_stream_name, subscribe_to: api_stream_name}
       ],
       strategy: :rest_for_one
     )
   end
 
-  # Parses the argument from the channel name, expecting a name formatted with
-  # `:` delimiters and `=` separators, e.g.
-  # `route=Red:direction_id=1:stop=place-sstat`
   @spec sses_opts(String.t()) :: Keyword.t()
-  defp sses_opts(key) do
-    filters =
-      table_keys(key)
-      |> Enum.map(fn {k, v} -> "filter[#{k}]=#{v}&" end)
-      |> Enum.join()
-
+  defp sses_opts(filters) do
     path =
-      "/predictions?#{filters}fields[prediction]=status,departure_time,arrival_time,direction_id,schedule_relationship,stop_sequence&include=route,trip,trip.occupancies,stop&fields[route]=long_name,short_name,type&fields[trip]=direction_id,headsign,name,bikes_allowed&fields[stop]=platform_code"
+      "/predictions?#{filters}&fields[prediction]=status,departure_time,arrival_time,direction_id,schedule_relationship,stop_sequence&include=route,trip,trip.occupancies,stop&fields[route]=long_name,short_name,type&fields[trip]=direction_id,headsign,name,bikes_allowed&fields[stop]=platform_code"
```

Comment:
question: while we're in here, are the `include` values here actually being used?